java/java.j2seplatform/src/org/netbeans/modules/java/j2seplatform/platformdefinition/J2SEPlatformDefaultJavadocImpl.java

```diff
@@ -116,6 +76,35 @@ public Collection<URI> getDefaultJavadoc(@NonNull final JavaPlatform platform) {
         return Collections.emptyList();
     }
 
+    private static String computeJavaDocURL(String version) {
+        switch (version) {
+            case "1.0": // NOI18N
+            case "1.1": // NOI18N
+            case "1.2": // NOI18N
+            case "1.3": // NOI18N
+            case "1.4": return null; // NOI18N
+            case "1.5": return "https://docs.oracle.com/javase/1.5.0/docs/api/"; // NOI18N
+            case "1.6": return "https://docs.oracle.com/javase/6/docs/api/"; // NOI18N
+            case "1.7": return "https://docs.oracle.com/javase/7/docs/api/"; // NOI18N
+            case "1.8": return "https://docs.oracle.com/javase/8/docs/api/"; // NOI18N
+        }
+        try {
+            int feature = Integer.parseInt(version);
+            if (feature >= 9) {
+                // timezone shouldn't matter since the accuracy is worse than a day
+                LocalDate jdk9 = LocalDate.of(2017, Month.SEPTEMBER, 21); // start of 6 month schedule
+                long max = ChronoUnit.MONTHS.between(jdk9, LocalDate.now()) / 6 + 9;
```

Comment:
For me the entire block is a little bit hard to understand. Especially the part where the monthly difference is divided by `6` and then `9` is added.
For example when Java 17 is used why the date (release date?!) of is relevant.

Reply:
It computes how many releases happened since JDK 9, then adds 9 to it.

Reply:
@mbien yes, it's clever, but still horrible! :laughing: Still think we should find a way to cache the results up to the source version supported by the IDE, then lazily compute any above that by pinging the links.

Reply:
1) what we know is that we don#t want to bump a number every release - thats how this all started, we already have to do that for the nb-javac lib, and we forget it literally every single time like in the current PR #6206 where I suspect that nobody will notice it until I point it out again :)
2) I think it is worse to compute the max known JDK version out of the NB version and probe the rest. Thats actually how I wanted to compute https://github.com/apache/netbeans/blob/8df65b70ed0d17f0e641b961b0ede8ef89ba640d/java/java.source.base/src/org/netbeans/modules/java/source/NoJavacHelper.java#L37 at first.

it has several issues. Results are not stable, they will depend on how well your wifi was when you start NB which will lead to "doc link missing -> can't reproduce" issues. It does also link two release schedules together. Taking one schedule and compute a value seems to be less complicated/direct approach.

The proper way in my opinion would be to ask the JDK what it is. If its EA it should produce an EA link, otherwise GA. This is also difficult to achieve since distributions don't use the same version scheme. This whole thing is a "best effort" approach anyway since the links never accounted for panama/loom builds etc and simply linked the wrong doc - same with JavaFX I suppose. Also: EA JDKs will receive GA doc if you are patient enough.

But if you really, really don't want the schedule approach, I would simply take the runtime version, use it as max known GA version and probe the rest (I hate pinging things since it introduces unpredictable latency into otherwise fast returning methods). Which is unstable, but doesn't link two schedules at least. I also don't want to access Runtime.version via reflection yet again tbh, I feel like every second module has to do this by now :)

Reply:
@mbien to be clear, I'm fine with date based for now.  A proper API that gives the maximum supported JDK by the current IDE configuration would be useful.  Mind you, maybe assuming everything up to `SourceVersion.latest()` is GA? Still, I do get your point about pinging.

Reply:
> and we forget it literally every single time like in the current PR 

What about a failing test? Runs quick, CI gets red and it's clear what's missing.

In general I'm not against the calculation approach, but I think it's quite hard to understand what's going for someone looking into that piece of code.

Just to ensure that it's clear what I'm talking about:

```java
           int javaVersion = Integer.parseInt(version);
            if (javaVersion >= 9) {

                int releasesAfterJava9 = javaVersion - 9;
                LocalDate releaseDateForThatJavaVersion = RELEASE_DATE_JAVA_9.plusMonths(JAVA_RELEASE_CYLCE_IN_MONTH * releasesAfterJava9);

                if (releaseDateForThatJavaVersion.isBefore(LocalDate.now())) {
                    return "https://docs.oracle.com/en/java/javase/" + javaVersion + "/docs/api/"; // NOI18N
                } else if (releaseDateForThatJavaVersion.isBefore(LocalDate.now().plusMonths(EARLY_ACCESS_PHASE_IN_MONTH))) {
                    return "https://download.java.net/java/early_access/jdk" + javaVersion + "/docs/api/"; // NOI18N
                }
            }
 ```

Reply:
> What about a failing test? Runs quick, CI gets red and it's clear what's missing.

@asbachb good idea, added some via #6308